src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs

```diff
@@ -773,6 +773,36 @@ private void Dispose()
                     }
                 }
             }
+            else if (reason == "watchpoint-trigger")
+            {
+                var wpt = results.Results.Find("wpt");
+                string bkptno = wpt.FindString("number");
+                ulong addr = cxt.pc ?? 0;
+
+                bool fContinue;
+                TupleValue frame = wpt.TryFind<TupleValue>("frame");
+                AD7BoundBreakpoint[] bkpt = _breakpointManager.FindHitBreakpoints(bkptno, addr, frame, out fContinue);
+                if (bkpt != null)
+                {
+                    List<object> bplist = new List<object>();
```

Comment:
I wouldn't think we should need to create the separate list. Can't we just do this?
new ReadOnlyCollection<object>(bkpt)